x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -135,18 +136,44 @@ export const signalRulesAlertType = ({
       logger.debug(buildRuleMessage(`interval: ${interval}`));
       await ruleStatusService.goingToRun();
 
-      const gap = getGapBetweenRuns({ previousStartedAt, interval, from, to });
+      let gap;
+      try {
+        gap = getGapBetweenRuns({ previousStartedAt, interval, from, to });
+      } catch (exc) {
+        const errorMessage = buildRuleMessage(`${exc.message} Check logs for further details.`);
+        logger.error(errorMessage);
+        await ruleStatusService.error(errorMessage, {
+          bulkCreateTimeDurations: result.bulkCreateTimes,
+          searchAfterTimeDurations: result.searchAfterTimes,
+          lastLookBackDate: result.lastLookBackDate?.toISOString(),
+        });
+        // throw the error because we can't query later on.
+        throw Error(exc.message);
```

Comment:
Throwing here to halt further execution. We could also put a return statement here but I think if the "from" param of the rule is NaN we should throw that as an error and not return "silently." I'm open to suggestions though.